### ISSUE TITLE
Evaluation timing verification

### ISSUE THREAD

**Issue by osrf-migration** [osrf/vrx-docker#13]
**Original report ([archived issue](https://osrf-migration.github.io/vrx-gh-pages/#!/osrf/vrx-docker/issues/13)) by Brian Bingham (Bitbucket: [brian_bingham](https://bitbucket.org/%7B9cdea257-248f-486e-8858-49d8485a916f%7D/)).**

----------------------------------------
Currently the run\_trial.bash script does the following in order:

* Starts up the vrx-server container
* Waits a fixed amount of time \(9 seconds\)
* Then starts the team’s container

We should verify that this timing won’t cause any problems.  I believe it is done this way b/c if we start the team’s container first, it will not have a ROS master on the network which could cause problems.

This seems to have the potential for the simulation to start \(and proceed towards Running state\) before the team’s code gets up and running, especially if the team’s container takes significant time to spin up.

Also, Tyler left notes in the source to check on this, so wanted to get it into our list.

‌

**Comment by M1chaelM** [comment]
This is a duplicate of issue #7, which includes Tyler's write up. Currently, the way we check for this is indirect, by making the testing infrastructure available to the teams and conducting the dress rehearsal.

We can avoid having to wait an arbitrary, fixed about of time by switching to Docker compose and using the depends_on flag. This is an easy way of making sure the team's container doesn't run until the vrx-server is up. It still doesn't prevent the competitor from starting while the team's container is getting ready.

There are some good suggestions for improving our approach here: https://docs.docker.com/compose/startup-order/

We can probably add a script that waits for the competitor container to be running before we start the `initial_state_duration` timer.
